### PR TITLE
Update superpmi_setup.py script with more exclusions

### PR DESCRIPTION
**Pull request by BruceForstall**
I noticed a bunch of failures trying to run crossgen2 on these binaries
in one of our collections.

Also, make the checking case-insensitive: I saw that we were failing to
match "corerun.exe" against "CoreRun.exe" that was in the exclusion list.

### CODE REVIEW

File: src/coreclr/scripts/superpmi_setup.py [hunk 1]
```diff
@@ -108,12 +152,15 @@
     "mscordbi.dll",
     "mscorrc.dll",
     "msdia140.dll",
+    "R2RDump.exe",
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi_setup.py:155]: This is not native binary, might be worth doing collection on this?

File: src/coreclr/scripts/superpmi_setup.py [hunk 2]
```diff
@@ -108,12 +152,15 @@
...
+    "R2RTest.exe",
     "superpmi.exe",
     "superpmi-shim-collector.dll",
     "superpmi-shim-counter.dll",
     "superpmi-shim-simple.dll",
     "System.IO.Compression.Native.dll",
     "ucrtbase.dll",
+    "xunit.console.exe",
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi_setup.py:163]: AFAIK, even same here. This must be a managed binary.

**Review by kunalspathak — COMMENTED** [review]
Added a question about native binaries.

File: src/coreclr/scripts/superpmi_setup.py (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/scripts/superpmi_setup.py:155]: I was going on the libraries.crossgen2.1.windows.x64.checked.log file from the last collection, where r2rdump.exe and xunit.console.exe failed:
```
[104:331]: C:\h\w\AE830971\p\superpmi\corerun.exe C:\h\w\AE830971\p\superpmi\crossgen2\crossgen2.dll @C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_R2RDump_R2RDump_exe_arusu6rw.rsp
'C:\h\w\AE830971\p\superpmi\corerun.exe C:\h\w\AE830971\p\superpmi\crossgen2\crossgen2.dll @C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_R2RDump_R2RDump_exe_arusu6rw.rsp': Error return code: 2
============== Contents of C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_R2RDump_R2RDump_exe_9h6l6pcd.stdout
No input files are loadable
============== End contents of C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_R2RDump_R2RDump_exe_9h6l6pcd.stdout
```

also:
```
[324:331]: C:\h\w\AE830971\p\superpmi\corerun.exe C:\h\w\AE830971\p\superpmi\crossgen2\crossgen2.dll @C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_xunit_core_dll_4bz1xglv.rsp
'C:\h\w\AE830971\p\superpmi\corerun.exe C:\h\w\AE830971\p\superpmi\crossgen2\crossgen2.dll @C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_xunit_console_exe_e8pdkc1s.rsp': Error return code: 2
============== Contents of C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_xunit_console_exe_w1nzc9yy.stdout
No input files are loadable
============== End contents of C:\h\w\AE830971\t\tmp_o4wsgj6\crossgen2_C__h_w_AE830971_w_AB3A0974_u_binaries_xunit_console_exe_w1nzc9yy.stdout
```

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_setup.py:155]: So here is the error we get in crossgen2 for `R2RDump.exe` and `R2RTest.exe`:

```
The format of a DLL or executable being loaded is invalid
```

And here is the error for `xunit.console.dll`:

```
The underlying compression routine could not be loaded correctly.
```

While these might be expected, I just want to double check with @davidwrighton . For this PR, I think you can go ahead and merge it.

File: src/coreclr/scripts/superpmi_setup.py [hunk 3]
```diff
@@ -202,7 +249,8 @@ def sorter_by_size(pair):
         # Credit: https://stackoverflow.com/a/19859907
         dirs[:] = [d for d in dirs if d not in exclude_directories]
         for name in files:
-            if name in exclude_files:
+            # Make the exclude check case-insensitive
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi_setup.py:252]: I would convert all the file names to lower case outside the loop:

```py
exclude_files_lower = [filename.lower() for filename in exclude_files]
for name in files:
     if name.lower() in exclude_files_lower:
          continue
```

**Review by kunalspathak — APPROVED** [review]
LGTM, with minor suggestion.

File: src/coreclr/scripts/superpmi_setup.py (hunk 3 above)
Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_setup.py:252]: I meant to have the `exclude_files_lower = [filename.lower() for filename in exclude_files]` outside of `for name in files:`

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_setup.py:252]: otherwise we are doing `.lower()` on `exclude_files` for every file we iterate.

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi_setup.py:252]: well, it's not perf critical...

**Pull request merged**